### PR TITLE
Add parametric url

### PR DESCRIPTION
**Pull request by py4mac**

### CODE REVIEW

File: hook.go [hunk 1]
```diff
@@ -147,3 +150,32 @@ func isInScope(ctx context.Context, s subscriber, scope string) bool {
...
+
+// ResolveUrl template from payload content
+func ResolveUrl(payload interface{}, URL *url.URL) *url.URL {
+	var tpl bytes.Buffer
+
+	urlCopy, _ := url.Parse(URL.String())
+
+	t, err := template.New("").Parse(urlCopy.RawQuery)
+	if err != nil {
+		return urlCopy
```
Comment (david972) [inline hook.go]: maybe add log here ?

File: hook.go [hunk 2]
```diff
@@ -147,3 +150,32 @@ func isInScope(ctx context.Context, s subscriber, scope string) bool {
...
+	}
+
+	var payloadAsMap map[string]interface{}
+	err = json.Unmarshal(payloadAsBin, &payloadAsMap)
+	if err != nil {
+		return urlCopy
+	}
+
+	_ = t.Execute(&tpl, payloadAsMap)
```
Comment (david972) [inline hook.go]: Why do not check the error here ?

File: hook.go (hunk 1 above)
Comment (py4mac) [reply hook.go]: done

File: hook.go (hunk 2 above)
Comment (py4mac) [reply hook.go]: added an error check

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,14 @@
 
-<a name="v0.2.1"></a>
-## [v0.2.1](https://gitlab.w6d.io/w6d/projectmgt/-/compare/v0.2.0...v0.2.1) (2022-01-28)
+<a name="0.2.2"></a>
```
Comment (david972) [inline docs/CHANGELOG.md]: minor version

**Review by david972 — CHANGES_REQUESTED** [review]

File: docs/CHANGELOG.md (hunk 3 above)
Comment (py4mac) [reply docs/CHANGELOG.md]: fixed

File: hook_test.go [hunk 4]
```diff
@@ -83,13 +117,18 @@ var _ = Describe("Hook", func() {
 				err := hook.Subscribe(context.Background(), "http://localhost", "[")
 				Expect(err).To(Succeed())
 				err = hook.DoSend(context.Background(), "message", "test")
-				Expect(err).ToNot(Succeed())
-				Expect(err.Error()).To(Equal("send failed"))
+				Expect(err).To(Succeed())
```
Comment (david972) [inline hook_test.go:120]: Why this test succeed ?
It shouldn’t as the scope isn’t a good regex

File: hook_test.go [hunk 5]
```diff
@@ -83,13 +117,18 @@ var _ = Describe("Hook", func() {
...
 			})
 			It("with DoSend", func() {
 				err := hook.DoSend(context.Background(), "message", "*")
+				Expect(err).To(Succeed())
```
Comment (david972) [inline hook_test.go:124]: Same as before

**Review by david972 — CHANGES_REQUESTED** [review]

File: hook_test.go (hunk 4 above)
Comment (py4mac) [reply hook_test.go:120]: We need to raise an error in case isInScope() condition is not respected
That's not the case here

Comment (david972) [reply hook_test.go:120]: Ok that means here is succeeded but not sent ?

Comment (py4mac) [reply hook_test.go:120]: yep, we can raise an error here https://github.com/w6d-io/hook/blob/ae14f0b4c93864d768011d3536d970c085bcf6e7/hook.go#L65

File: hook_test.go (hunk 5 above)
Comment (py4mac) [reply hook_test.go:124]: It's different here as there is no subscriber 
See https://github.com/w6d-io/hook/blob/ae14f0b4c93864d768011d3536d970c085bcf6e7/hook_test.go#L102

**Pull request merged**